src/frontend/src/binder/relation/join.rs

```diff
@@ -107,7 +111,52 @@ impl Binder {
         Ok(match constraint {
             JoinConstraint::None => (ExprImpl::literal_bool(true), None),
             JoinConstraint::Natural => {
-                return Err(ErrorCode::NotImplemented("Natural join".into(), 1633.into()).into())
+                // First, we identify columns with the same name.
+                let old_context = self.context.clone();
+                self.push_lateral_context();
+                // Bind this table factor to an empty context
+                let table_factor = table_factor.unwrap();
+                let right_table = get_table_name(&table_factor);
```

Comment:
FYI using `get_table_name` on right_table_factor would not work once we support nested join.

For example, in `a natural join (b join c on true)`, we need to find the common columns between `a` and `(b join c on true)`. The right part of join does not even have a single table name, and the whole `get_table_name` was just a hack/workaround for `join-using`. `join-using` would also break after nested join.

Reply:
Actually, the more I think of it, we could recursively resolve the source table names in nested join.

Reply:
Anyway, this is not relevant to the PR, we will solve it in nested join PR